File: fastApiProject/entity/GuessRecord.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import datetime


class GuessRecord(BaseModel):
    """猜字游戏猜测记录实体"""
    id: Optional[int] = Field(None, description="记录唯一ID")
    guess_word_id: int = Field(..., description="关联的目标字ID")
    guess: str = Field(..., description="猜测的字/词")
    similarity: float = Field(..., description="相似度:0-1")
    is_correct: bool = Field(False, description="是否猜中")
    created_at: Optional[datetime] = Field(None, description="猜测时间")

    class Config:
        from_attributes = True

File: blog-agent/agent/src/main/java/com/xiafan/agent/entity/GuessRecord.java
package com.xiafan.agent.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.Data;
import lombok.Getter;

import java.time.LocalDateTime;

/** guess_records */
@TableName("guess_records")
@Data
public class GuessRecord {
    @TableId(value = "id", type = IdType.AUTO)
    private Integer id;
    @TableField("guess_word_id")
    private Integer guessWordId;
    @TableField("guess")
    private String guess;
    @TableField("similarity")
    private Double similarity;

    @TableField("is_correct")
    @Getter(onMethod_ = @JsonProperty("is_correct"))
    private boolean isCorrect = false;

    @TableField("created_at")
    private LocalDateTime createdAt;
}
